From a3f5a431fd7c4b5d476922bbc33c6ba98b9513ac Mon Sep 17 00:00:00 2001
From: Andrii Osipov <aosipov@anaconda.com>
Date: Thu, 2 Jul 2026 03:21:18 -0700
Subject: [PATCH] remove -z and -pic on osx for tests

---
 configure.ac      |  6 ++++++
 tests/Makefile.am | 16 ++++++++++++++++
 2 files changed, 22 insertions(+)

diff --git a/configure.ac b/configure.ac
index fee6302..3830c4b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -36,6 +36,12 @@ AM_CONDITIONAL([WITH_ASAN], [test x"$with_asan" = xyes])
 
 AX_CXX_COMPILE_STDCXX([17], [noext], [])
 
+case $host_os in
+  darwin*) is_darwin=yes ;;
+  *)       is_darwin=no ;;
+esac
+AM_CONDITIONAL([OS_DARWIN], [test "x$is_darwin" = xyes])
+
 AC_ARG_WITH([ubsan],
    AS_HELP_STRING([--with-ubsan], [Build with undefined behavior sanitizer])
 )
diff --git a/tests/Makefile.am b/tests/Makefile.am
index 8e11d45..fa87e9d 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -85,22 +85,34 @@ clean-local:
 
 # by default, use -fpic to compile
 AM_CFLAGS = -fpic
+if OS_DARWIN
+LDFLAGS_local = -L. $(AM_LDFLAGS)
+LDFLAGS_sharedlib = -shared -L. $(AM_LDFLAGS)
+else
 LDFLAGS_local = -Wl,--disable-new-dtags -Wl,-rpath-link=. -L. $(AM_LDFLAGS)
 LDFLAGS_sharedlib = -Wl,--disable-new-dtags -shared -L. $(AM_LDFLAGS)
+endif
 export NIX_DONT_SET_RPATH=1
 export NIX_LDFLAGS=
 
 simple_SOURCES = simple.c
 # no -fpic for simple.o
 simple_CFLAGS =
+if OS_DARWIN
+simple_LDFLAGS =
+else
 simple_LDFLAGS = -Wl,-z,noexecstack
+endif
+
 
 simple_pie_SOURCES = simple.c
 simple_pie_CFLAGS = -fPIC -pie
 
+if !OS_DARWIN
 simple_execstack_SOURCES = simple.c
 simple_execstack_CFLAGS =
 simple_execstack_LDFLAGS = -Wl,-z,execstack
+endif
 
 main_SOURCES = main.c
 main_LDADD = -lfoo $(AM_LDADD)
@@ -112,7 +124,11 @@ main_no_pie_SOURCES = main.c
 main_no_pie_CFLAGS = -fno-pie
 main_no_pie_LDADD = -lfoo $(AM_LDADD)
 main_no_pie_DEPENDENCIES = libfoo.so
+if OS_DARWIN
+main_no_pie_LDFLAGS = $(LDFLAGS_local) -Wl,-no_pie
+else
 main_no_pie_LDFLAGS = $(LDFLAGS_local) -no-pie
+endif
 
 # Non-PIE binary with a max-page-size larger than the runtime page size. On
 # 32-bit targets this makes patchelf relocate .dynamic into the read-only first
-- 
2.39.5 (Apple Git-154)

